security/aws/iam_expired_ssl_certs/iam_expired_ssl_certs.pt

```diff
@@ -0,0 +1,153 @@
+long_description ""
+category "Security"
+severity "high"
+default_frequency "daily"
+info(
+  version: "2.0",
+  provider: "AWS",
+  service: "IAM",
+  policy_set: "CIS",
+  cce_id: "", # No associated CCE
+  cis_aws_foundations_securityhub: "", # Unlisted
+  benchmark_control: "1.19",
+  benchmark_version: "1.4.0",
+  cis_controls: "[\"3.1v8\", \"13v7\"]",
+  nist: "" # Unknown
+)
+
+###############################################################################
+# User inputs
+###############################################################################
+
+parameter "param_email" do
+  type "list"
+  label "Email Address"
+  description "Email addresses of the recipients you wish to notify"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+credentials "auth_aws" do
+  schemes "aws","aws_sts"
+  label "AWS"
+  description "Select the AWS Credential from the list"
+  tags "provider=aws"
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+# Retrieve the id number of the account being tested
+datasource "ds_get_caller_identity" do
+  request do
+    auth $auth_aws
+    verb "GET"
+    host "sts.amazonaws.com"
+    path "/"
+    header "User-Agent", "RS Policies"
+    query "Action", "GetCallerIdentity"
+    query "Version", "2011-06-15"
+  end
+  result do
+    encoding "xml"
+    collect xpath(response, "//GetCallerIdentityResponse/GetCallerIdentityResult") do
+      field "account",xpath(col_item, "Account")
+    end
+  end
+end
+
+datasource "ds_iam_list_server_certs" do
+  request do
+    auth $auth_aws
+    verb "GET"
+    host "iam.amazonaws.com"
+    path "/"
+    query "Action", "ListServerCertificates"
+    query "Version", "2010-05-08"
+    header "Accept", "application/json"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    field "certificates", jmes_path(response, "ListServerCertificatesResponse.ListServerCertificatesResult.ServerCertificateMetadataList")
+  end
+end
+
+datasource "ds_iam_server_certs" do
+  run_script $js_server_certs, $ds_get_caller_identity, $ds_iam_list_server_certs
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_server_certs", type:"javascript" do
+  parameters "ds_get_caller_identity", "ds_iam_list_server_certs"
+  result "result"
+  code <<-EOS
+    var now = new Date()
+    var bad_cert_list = []
+
+    _.each(ds_iam_list_server_certs.certificates, function(cert){
+      var expiration = new Date(1000 * cert.Expiration);
+      if (now > expiration) {
+        bad_cert_list.push(cert.Arn + " expired on " + expiration.toString())
```

Comment:
can we make this { arn: "", expiration:"", account: ""} so we can iterate them in the export?